src/main/java/com/conveyal/analysis/controllers/BundleController.java

```diff
@@ -135,6 +134,13 @@ private Bundle create (Request req, Response res) {
                 bundle.feedsComplete = bundleWithFeed.feedsComplete;
                 bundle.totalFeeds = bundleWithFeed.totalFeeds;
             }
+            if (files.get("config") != null) {
+                // For validation, rather than reading as freeform JSON, deserialize into a model class instance.
+                // However, only the instance fields specifying things other than OSM and GTFS IDs will be retained.
+                // Use strict object mapper (from the strict/lenient pair) to fail on misspelled field names.
+                String configString = files.get("config").get(0).getString();
+                bundle.config = JsonUtilities.objectMapper.readValue(configString, TransportNetworkConfig.class);
```

Comment:
Do we want actually want strict validation of the JSON done here? If we have a new version of the worker that uses new config values, we would need to deploy a new backend with a new TransportNetworkConfig containing those new values in order to create bundles using those new config values. We lose part of the benefit of using a freeform JSON field in that all allowable fields must be predefined in the backend at the time of deployment.

I already ran into this while creating the UI component, because the proposed defaults contain a value that is not currently in the config.

Reply:
Very good point. We've consistently tried to allow single-purpose or experimental worker versions to function without the backend understanding their particularities, and we need to do the same here. To that end, any validation of specific features should be done by the worker that builds the network, not by the backend. I do want to perform validation because it could be very confusing (and lead to trivial support requests and degraded user experience) if small faults in the configuration were accepted and silent failures ensued.

The backend should perform as much validation as it can in order to fail fast, so someone doesn't have to move on to starting a worker and running some analysis to discover a simple mistake in JSON quoting (for example). Any new special-purpose worker version should be adding fields to those already known by the backend. So I think we want the backend to validate that it's JSON and any fields it already knows about can be deserialized properly, while ignoring the fields it doesn't know about. That's the behavior I originally had, and I shouldn't have added the second commit to alter it.

Reply:
On further thought, even on the worker we always need to ignore unknown field names. There are legitimate reasons a network built with the config might later be used with an older worker that doesn't possess the new feature (if only for testing the effects of the new feature). This has been clarified in the code comments.